utils/prettier.js

```diff
@@ -0,0 +1,184 @@
+  }
+  return beforePrettierState;
+};
+
+/**
+ * HANDLING OF VARIOUS FILE TYPES
+ * `parserScriptUrl`s from https://unpkg.com/browse/prettier@1.13.0/
+ * `parserName`s from https://prettier.io/docs/en/options.html#parser
+ */
+
+const prettierParserMap = {
+  css: {
+    parserScriptUrl: 'https://unpkg.com/prettier@1.13.0/parser-postcss.js',
+    parserName: 'css',
+  },
+  js: {
+    parserScriptUrl: 'https://unpkg.com/prettier@1.13.0/parser-babylon.js',
+    parserName: 'babylon',
+  },
+  json: {
+    parserScriptUrl: 'https://unpkg.com/prettier@1.13.0/parser-babylon.js',
+    parserName: 'json',
+  },
+  // TODO: uncomment markdown entry after addressing issue no. 181
+  // https://github.com/FormidableLabs/runpkg/issues/181
+  // md: {
+  //   parserScriptUrl: 'https://unpkg.com/prettier@1.13.0/parser-markdown.js',
+  //   parserName: 'markdown',
+  // },
+  less: {
+    parserScriptUrl: 'https://unpkg.com/prettier@1.13.0/parser-postcss.js',
+    parserName: 'css',
+  },
+  map: {
+    // e.g. `vuetify@2.2.15/lib/components/VDatePicker/VDatePickerDateTable.js.map`
+    parserScriptUrl: 'https://unpkg.com/prettier@1.13.0/parser-babylon.js',
+    parserName: 'json',
+  },
+  mjs: {
+    parserScriptUrl: 'https://unpkg.com/prettier@1.13.0/parser-babylon.js',
+    parserName: 'babylon',
+  },
+  sass: {
+    parserScriptUrl: 'https://unpkg.com/prettier@1.13.0/parser-postcss.js',
+    parserName: 'css',
+  },
+  scss: {
+    parserScriptUrl: 'https://unpkg.com/prettier@1.13.0/parser-postcss.js',
+    parserName: 'css',
+  },
+  ts: {
+    parserScriptUrl: 'https://unpkg.com/prettier@1.13.0/parser-typescript.js',
+    parserName: 'typescript',
+  },
+};
+
+/**
+ * HANDLING PRETTIER FORMAT BUTTON PRESS
+ */
+
+const loadPrettierParserScriptForExtension = ({
+  fileData,
+  setPrettierButtonState,
+  dispatch,
+  request,
+}) => {
+  const trueExt = getFinalExtension(fileData.extension);
+  const parserLangConfig = prettierParserMap[trueExt];
+
+  if (!parserLangConfig) {
+    // This *should* never happen thanks to `pickInitialPrettierState`
+    console.error(
+      'File extension not supported. Prettier button should be hidden/disabled'
+    );
+    setPrettierButtonState(cannotPrettierState);
+    return;
+  }
+
+  const scriptUrl = parserLangConfig.parserScriptUrl;
+
+  const isScriptAlreadyPresent = [
+    ...document.querySelectorAll('head script'),
+  ].some(script => script.src === scriptUrl);
+
+  const tryToPrettify = () => {
+    try {
+      const code = prettier.format(fileData.code, {
+        parser: parserLangConfig.parserName,
+        plugins: prettierPlugins,
+      });
+      dispatch({
+        type: 'setCache',
+        payload: {
+          ['https://unpkg.com/' + request.path]: {
+            ...fileData,
+            code,
+          },
+        },
+      });
+      setPrettierButtonState(afterPrettierState);
+    } catch (e) {
+      console.error(e);
+      setPrettierButtonState(errorPrettierState);
```

Comment:
💭 put the button state setting in the UI component? better separation of concerns?

Reply:
💭 right after the call to `loadPrettierParserScriptForExtension`?
https://github.com/FormidableLabs/runpkg/pull/184/files#diff-bb8820440d0617732459bfe856f864abR42

Reply:
I'm 50/50 about this! On one hand it could be nice to keep the `setState`s in the component where the `useState` is used instead of passing the setter to this file, but on the other hand I feel this way keeps the `Article` component less cluttered (and is a bit easier). 

Open to going the route you suggest though 🤔. Might see what others think